src/main/webapp/js/login.js

```diff
@@ -0,0 +1,384 @@
+// limitations under the License.
+
+/* ==========================================================================
+   USER SIGN-IN
+   ========================================================================== */
+function onSignIn(googleUser) {
+    let id_token = googleUser.getAuthResponse().id_token;
+    let profile = googleUser.getBasicProfile();
+    fetch(`/login?id_token=${id_token}`)
+        .then((response) => response.json())
+        .then((data) => {
+            localStorage.setItem("user", data.sub);
+            localStorage.setItem("loggedIn", true);
+            addUserContent(profile.getName(), profile.getImageUrl());
+            showAccountMenu();
+        }).catch((error) => {
+            console.log(error);
+        });
+}
+
+//Add user information to signed in UI
+function addUserContent(name, image) {
+    // document.getElementById("user-name").innerText = name;
+    document.getElementById("profile-pic").src = image;
+}
+
+//Replaces the sign-in button with signed in UI
+function showAccountMenu() {
+    document.getElementById("account-menu").classList.add("show");
+    document.getElementById("account-menu").classList.remove("hide");
+    document.getElementById("sign-in").classList.add("hide");
+    document.getElementById("sign-in").classList.remove("show");
+}
+
+function hideAccountMenu() {
+    document.getElementById("account-menu").classList.add("hide");
+    document.getElementById("account-menu").classList.remove("show");
+    document.getElementById("sign-in").classList.add("show");
+    document.getElementById("sign-in").classList.remove("hide");
+}
+
+//Logs out of the account
+function signOut() {
+    var auth2 = gapi.auth2.getAuthInstance();
+    auth2.signOut().then(function() {
+        console.log('User signed out.');
+    });
+    localStorage.setItem("user", 0);
+    hideAccountMenu();
+}
+
+/*
+    FUNCTIONS RELATED TO THE ACCOUNTS PAGE
+*/
+
+function accountFunctions() {
+    getNumSearches();
+    getLastVisited();
+    getFavFood();
+    getNumReviews();
+}
+
+function getNumSearches() {
+    let count = 0;
+    let numSearchesEl = document.getElementById('num-searches');
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((searches) => {
+        searches.forEach((search) => {
+            count += 1;
+        });
+        numSearchesEl.innerText = count;
+    });
+}
+
+function getLastVisited() {
+    let lastVisitedEl = document.getElementById('last-visited');
+    if (localStorage.getItem("loggedIn")) {
+        userID = localStorage.getItem("user");
+    }
+    fetch(`/searches?user=${userID}`, { method: 'GET' }).then(response => response.json()).then((searches) => {
+        for (search of searches) {
+            lastVisitedEl.innerText = search.name;
+            break;
```

Comment:
this is a for loop that loops once and then breaks. is there a better way to do this?

Reply:
done